domain-mapping/api/class-darkmatter-domains.php

```diff
@@ -160,6 +201,7 @@ public function add( $fqdn = '', $is_primary = false, $is_https = false, $force
 			'domain'     => $fqdn,
 			'is_primary' => ( ! $is_primary ? false : true ),
 			'is_https'   => ( ! $is_https ? false : true ),
+			'type'       => ( ! empty( $type ) ? $type : DM_DOMAIN_TYPE_MAIN ),
```

Comment:
Missing type validation, I would refactor the relevant code from `get_domains_by_type`